src/services/Game/services/BallMovementChecker.ts

```diff
@@ -0,0 +1,221 @@
+import TObjectLiteral from 'types/TObjectLiteral';
+
+import { GameConstants } from '../contants';
+import { TLvlCreaser } from '../types';
+
+export class BallMovementChecker {
+    map: string[];
+
+    constructor(map: string[], richedKeys: TObjectLiteral) {
+        this.map = map;
+
+        Object.entries(richedKeys).forEach(([key, value]) => {
+            if (value) {
+                this.convertCurrentBlockType(key.toUpperCase());
+            }
+        });
+    }
+
+    public emptyBlocks = ['0', '-', '+'];
+
+    public filledBlock = ['1', 'a', 'b', 'c', 'A', 'B', 'C'];
+
+    checkNextLvlRiched(ballX: number, ballY: number): TLvlCreaser | null {
+        const ballCenterX = ballX + GameConstants.GAMER_DIAMETR / 2;
+        const ballCenterY = ballY + GameConstants.GAMER_DIAMETR / 2;
+
+        const blockX = Math.floor(ballCenterX / GameConstants.PERFECT_ONE);
+        const blockY = Math.floor(ballCenterY / GameConstants.PERFECT_ONE);
+
+        const blockValue = this.map[blockY].charAt(blockX);
+
+        if (blockValue === '-' || blockValue === '+') {
+            return blockValue;
+        }
+
+        return null;
+    }
+
+    getBlockUnderBallValue(ballX: number, ballY: number): string {
+        const ballCenterX = ballX + GameConstants.GAMER_DIAMETR / 2;
+        const ballCenterY = ballY + GameConstants.GAMER_DIAMETR;
+
+        const blockX = Math.floor(ballCenterX / GameConstants.PERFECT_ONE);
+        const blockY = Math.floor(ballCenterY / GameConstants.PERFECT_ONE);
+
+        return this.map[blockY].charAt(blockX);
+    }
+
+    getBlockWasPressed(
+        ballX: number,
+        ballY: number,
+        velY: number,
+    ): string | null {
+        const blockValue = this.getBlockUnderBallValue(
+            ballX,
+            ballY + GameConstants.GAMER_DIAMETR / 4,
+        );
+
+        if (
+            blockValue.toLowerCase() !== blockValue.toUpperCase()
+            && blockValue.toLowerCase() === blockValue
+            && Number.isNaN(parseInt(blockValue, 10))
+            && velY !== 0
+        ) {
+            this.convertCurrentBlockType(blockValue.toLocaleUpperCase());
+
+            return blockValue;
+        }
+
+        return null;
+    }
+
+    isBallCanFall(ballX: number, ballY: number): boolean {
+        // eslint-disable-next-line no-param-reassign
+        ballX = Math.round(ballX * 100000000) / 100000000;
+        // eslint-disable-next-line no-param-reassign
+        ballY = Math.round(ballY * 100000000) / 100000000;
```

Comment:
🧐
Давай сохраним как-то так
```suggestion
        const ballCoords = { x: ..., y: ... }
```
Тут нет причин прибивать проверки линтера

Reply:
И да, в чём прикол этой операции?

Reply:
Переделал. Это округление) 
Чтобы рассчет в миллиарды миллиардных не скатывались